core/models/machine_request.py

```diff
@@ -297,6 +297,8 @@ def get_credentials(self):
         old_admin = old_provider.get_admin_identity().get_credentials()
         if 'ex_force_auth_version' not in old_creds:
             old_creds['ex_force_auth_version'] = '2.0_password'
+        if old_creds['ex_force_auth_version'] != '2.0_password' and 'domain_name' not in old_creds:
```

Comment:
Note: I had to read through this method a couple of times. The name (`get_credentials`) is a bit deceptive. It does quite a bit more than just 'getting'. It seems to do some clever copying and manipulation. I imagine it started out simpler.